### PR TITLE
Refactor the httpclient out of the aiia service. Split out the API surface into a different file too.

### PR DESCRIPTION
**Pull request by dacspiir**
Note that there are a number of TODOs left around

### CODE REVIEW

File: Web/AiiaClient/AiiaHttpClient.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        public async Task<TResponse> CallApi<TRequest, TResponse>(string url,
+            TRequest body,
+            HttpMethod method,
+            string? authScheme,
+            string? authToken)
+        {
+            // prepare request
+            var httpRequestMessage = new HttpRequestMessage(method, url);
+
+            if (body is not null)
```
Comment (adzspiir) [inline Web/AiiaClient/AiiaHttpClient.cs:33]: i thought that 'is not null' is only allowed in c# 9.0, but isn't this project still in c#? i mean if it works it works, im just confused a bit

File: Web/AiiaClient/AiiaClientSecrets.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Aiia.Sample.Services;
+
+public struct AiiaClientSecrets
```
Comment (adzspiir) [inline Web/AiiaClient/AiiaClientSecrets.cs:3]: can this (AiiaClientSecrets) and AiiaAccessTokens get renamed to singular names?

**Review by adzspiir — CHANGES_REQUESTED** [review]
looks really good :) can we run a clean-up on this before merging?

File: Web/AiiaClient/Aiia/Payment.cs [hunk 3]
```diff
@@ -1,10 +1,10 @@
-using NodaTime;
+
 
 namespace Aiia.Sample.Models.Aiia
 {
     public class Payment
     {
-        public Instant CreatedAt { get; set; }
+        public string CreatedAt { get; set; }
```
Comment (dacspiir) [inline Web/AiiaClient/Aiia/Payment.cs:7]: TODO: this was Instant

File: Web/AiiaClient/Aiia/PaymentAuthorization.cs [hunk 4]
```diff
@@ -1,10 +1,9 @@
-using NodaTime;
 
 namespace Aiia.Sample.Models.Aiia
 {
     public class PaymentAuthorization
     {
-        public Instant CreatedAt { get; set; }
+        public string CreatedAt { get; set; }
```
Comment (dacspiir) [inline Web/AiiaClient/Aiia/PaymentAuthorization.cs:6]: TODO: this was Instant

File: Web/AiiaClient/Aiia/PaymentsQueryRequest.cs [hunk 5]
```diff
@@ -1,11 +1,10 @@
 using System.Collections.Immutable;
-using NodaTime;
 
 namespace Aiia.Sample.Models.Aiia
 {
     public class PaymentsQueryRequest
     {
-        public Interval? Interval { get; set; }
+        public string Interval { get; set; }
```
Comment (dacspiir) [inline Web/AiiaClient/Aiia/PaymentsQueryRequest.cs:7]: TODO: this was Instant?

File: Web/AiiaClient/Aiia/TransactionDetails.cs [hunk 6]
```diff
@@ -1,17 +1,16 @@
-using NodaTime;
 
 namespace Aiia.Sample.Models.Aiia
 {
     public class TransactionDetails
     {
         public CurrencyConversionViewModel CurrencyConversion { get; set; }
         public TransactionPartyViewModel Destination { get; set; }
-        public LocalDate? ExecutionDate { get; set; }
+        public string? ExecutionDate { get; set; }
```
Comment (dacspiir) [inline Web/AiiaClient/Aiia/TransactionDetails.cs:8]: TODO: this was LocalDate?

File: Web/AiiaClient/Aiia/TransactionDetails.cs [hunk 7]
```diff
@@ -1,17 +1,16 @@
...
         public TransactionIdentifiersViewModel Identifiers { get; set; }
         public string Message { get; set; }
         public RewardViewModel Reward { get; set; }
         public TransactionPartyViewModel Source { get; set; }
-        public LocalDate? ValueDate { get; set; }
+        public string? ValueDate { get; set; }
```
Comment (dacspiir) [inline Web/AiiaClient/Aiia/TransactionDetails.cs:13]: TODO: this was LocalDate?

File: Web/AiiaClient/AiiaHttpClient.cs (hunk 1 above)
Comment (dacspiir) [reply Web/AiiaClient/AiiaHttpClient.cs:33]: I thought it wouldn't work too
I did set the language version to 'latest' and it just compiles and runs 🤷

File: Web/AiiaClient/AiiaApi.cs [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        {
+            return await _aiiaHttpClient.HttpGet<InboundPayment>($"v1/accounts/{accountId}/payments/inbound/{paymentId}",accessTokens);
+        }
+
+        public async Task<PayerTokenModel> GetInboundPaymentPayerToken(AiiaAccessTokens accessTokens, string accountId,
+            string paymentId)
+        {
+            return await _aiiaHttpClient.HttpPost<object, PayerTokenModel>(
+                $"v1/accounts/{accountId}/payments/inbound/{paymentId}/payer-token", new object(), accessTokens);
+
```
[the remark this answers is not in the document]
Comment (adzspiir) [reply Web/AiiaClient/AiiaApi.cs:92]: bump

File: Web/AiiaClient/AiiaClientSecrets.cs (hunk 2 above)
Comment (adzspiir) [reply Web/AiiaClient/AiiaClientSecrets.cs]: bump

**Review by adzspiir — APPROVED** [review]

**Pull request merged**